var/spack/repos/builtin/packages/py-submitit/package.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PySubmitit(PythonPackage):
```

Comment:
Good to see. Will we "converge" on this in any way, or is this just the N+1 of submission mechanisms?

Reply:
(also, you could file this package upstream, too :))

Reply:
Converging would be surely good, however at the moment I know these mechanism being used:
- In `circuit-build`, call the `bbp_env` function to build a command string including `salloc srun` and use `snakemake` to run it.
- In `bbp-workflow`, use the functions defined in `slurm.py` and use `luigi`.

Do you know other ways that we are using to submit jobs to slurm from python?

In the current use case, using a library like `submitit` seems simpler because there is no need to include a full workflow manager like `snakemake` or `luigi`, but a layer that separates the slurm commands from the main logic is useful.
The code using the library is contained in a single function of a dozen of lines, so it should be relatively easy to change it if we need.

Reply:
(PR created in upstream using pypi instead of url. This file can be overwritten when the spack version will be updated)

Reply:
Thanks for filing this upstream!

I was more thinking in terms of parallelizing libraries… there's also all of `dask-*`, `ipyparallelpy-*`…

Reply:
Oh I see, thanks for clarifying. The approach used in this case aims to reuse the existing script (s2f-recipe) and call it independently for different regions. Each job is already parallelized using multiple processes with joblib.
Using dask would have required a significative refactoring (and probably new issues to resolve).